src/js/components/VariableMeta.js

```diff
@@ -8,99 +8,87 @@ import {
+        container.innerHTML = JSON.stringify(src, null, '  ');
 
-            this.state.clipboard.on('success', (e) => {
-                this.setState({copy_state: 'success'});
-            });
+        document.body.appendChild(container);
+        container.select();
+        document.execCommand('copy');
 
-            this.state.clipboard.on('error', (e) => {
-                this.setState({copy_state: 'error'});
+        document.body.removeChild(container);
+
+        this.copiedTimer = setTimeout(() => {
+            this.setState({
+                copied: false
             });
-        }
+        }, 5500)
 
-        this.state.copy_state = null;
-    }
+        this.setState({ copied: true }, () => {
+            if (typeof enableClipboard !== "function") {
+                return;
+            }
 
-    componentWillUnmount = () => {
-        this.state.clipboard && this.state.clipboard.destroy();
+            enableClipboard({
+                src:src,
+                namespace: namespace,
+                name: namespace[namespace.length - 1]
+            });
+        })
     }
 
     getCopyComponent = () => {
-        const {id, copy_state} = this.state;
         const {src, size, theme, enableClipboard} = this.props;
-
         let style = Theme(theme, 'copy-to-clipboard').style;
+        
         return (
             enableClipboard ?
             <span class="copy-to-clipboard-container" >
                 <span
                 style={{
                     ...style,
-                    display:copy_state=='success' ? 'none' : 'inline-block'
+                    display: 'inline-block'
                 }}
-                data-clipboard-text={JSON.stringify(src, null, '  ')}
-                id={"clipboard-container-" + id} >{this.getClippyIcon()}</span>
-                <span
-                style={{
-                    ...style,
-                    display:copy_state == 'success'
-                    ? 'inline-block' : 'none'
-                }} >{this.getClippyIcon()}</span>
+                onClick={this.handleCopy}
+                >{this.getClippyIcon()}</span>
             </span>
             : null
         );
     }
 
     getClippyIcon = () => {
-        const {enableClipboard, theme, src, namespace} = this.props;
-        if (typeof enableClipboard === "function") {
-            return <Clippy class="copy-icon"
-                {...Theme(theme, 'copy-icon')}
-                onClick={() => {
-                    enableClipboard({
-                        src:src,
-                        namespace: namespace,
-                        name: namespace[namespace.length - 1]
-                    })
-                }}
-            />
-        } else {
-            return <Clippy class="copy-icon" {...Theme(theme, 'copy-icon')} />
+        const {theme} = this.props;
+
+        if (this.state.copied) {
+            return (<span>
+                <Clippy class="copy-icon" {...Theme(theme, 'copy-icon')} />
```

Comment:
it would be cool to render an entirely different icon here.  it's tough to see the check mark behind the clipboard icon

Reply:
after testing a little more, i think it's just some themes that make the check hard to see

Reply:
I didn't know if you were picking from a particular iconset or not, I think ideally it would be an icon that has the clipboard in it with some kind of affirmation symbol (checkmark or otherwise) in front of it. At the very least I'll picking another style here for the color options and test it out under more themes.